panels/user-accounts/um-password-dialog.c

```diff
@@ -223,6 +224,14 @@ accept_password_dialog (GtkButton        *button,
                         act_user_set_automatic_login (um->user, FALSE);
                         break;
 
+                case ACT_USER_PASSWORD_MODE_NONE:
+                        /* When setting the user password mode to none, when it
+                         * was previously set-at-login, we need to reset the password,
+                         * otherwise the mode will be set back to set-at-login */
+                        act_user_set_password (um->user, "", "");
+                        act_user_set_password_mode (um->user,  um->password_mode);
```

Comment:
Well, that makes the code more orthogonal without a much improved readability IMO: if you have to change the case later you also have to change that call. No need for repetition.